### ISSUE TITLE
Update pkgorigins.version on code changes?

### ISSUE THREAD

**Issue by IanButterworth** [timholy/Revise.jl#684]
With the addition of the [version field to pkgorigins](https://github.com/JuliaLang/julia/pull/45607) and [the API to access it](https://github.com/JuliaLang/julia/pull/45607), I wondered whether Revise should monitor for package version changes and update appropriately?

If so, are changes to the Project.toml itself tracked by Revise, or would the re-parsing of the TOML need to happen on any code change?

cc. @KristofferC

**Comment by timholy** [comment]
Yes, when an update is successful this would make a lot of sense.

**Comment by IanButterworth** [comment]
> You mean Revise should take care of updating this dict when it loads new pkg versions?

Yeah

**Comment by IanButterworth** [comment]
I couldn't figure out the right place to do it, but it needs something like this done any time the `Project.toml` changes, which I guess is a file that isn't currently watched? 

It also would need the package id (`id` here) which _could_ be constructed from the Project.toml, but that might not be healthy design?
```
    origin = get(Base.pkgorigins, id, nothing)
    if origin isa Base.PkgOrigin && !isnothing(origin.path)
        package_root = dirname(dirname(origin.path)) # assumes `path` is src/Foo.jl
        origin.version = Base.get_pkgversion_from_path(package_root)
    else
        @debug "Failed to update package version" id origin
    end
```

**Comment by IanButterworth** [comment]
Pkg will soon start highlighting when the pkgorigins version entry for a loaded dependency doesn't match the expected manifest version, so this needs fixing, otherwise it could be confusing. https://github.com/JuliaLang/Pkg.jl/pull/4109

**Comment by timholy** [comment]
Ugh, yet again git weirdness: https://github.com/timholy/claude_config/pull/12
